src/engine/GroupByImpl.h

```diff
@@ -28,6 +28,11 @@
 // Block size for when using the hash map optimization
 static constexpr size_t GROUP_BY_HASH_MAP_BLOCK_SIZE = 262144;
 
+namespace groupBy::detail {
+template <size_t IN_WIDTH, size_t OUT_WIDTH>
+class LazyGroupByRange;
+}
```

Comment:
Is this forward declaration necessary? I thought `friend class` would already do this implicitly?